commands/provider/service_deployer.go

```diff
@@ -2,41 +2,68 @@ package provider
 
 import (
 	"context"
+	"errors"
 	"io"
 
 	"github.com/asaskevich/govalidator"
 	"github.com/docker/docker/pkg/archive"
 	"github.com/mesg-foundation/core/protobuf/coreapi"
 )
 
+// StatusType indicates the type of status message.
+type StatusType int
+
+const (
+	// RUNNING indicates that status message belongs to an active state.
+	RUNNING StatusType = iota + 1
```

Comment:
One small change - use `iota` - there is no obvious reason why counting should be started from 1.

Reply:
in goi it's a best practice to count constants from 1. i guess this due to 0 is a _zero_ value in go

Reply:
Can you provide some article/best practice about this? I want to read about it.
thanks

Reply:
https://github.com/golang/go/wiki/Iota

Reply:
We could add a status `UNKNOWN` and start at `iota` like that it's ok. I agree that we shouldn't start at 0, we can have some side effects with that.

Like here https://github.com/mesg-foundation/core/blob/a6887f298acf9e1e1d09f6b853d44a660b3b3a46/service/status.go#L11-L17

or here

https://github.com/mesg-foundation/core/blob/a6887f298acf9e1e1d09f6b853d44a660b3b3a46/container/type.go#L7-L12